### PR TITLE
feat(chartGraph): convert graph data, add error/loading states

### LINKED ISSUE #18: Dashboard graph card helpers
## Feature description

As a developer I'd like to offset functionality of the Graph Card display into a series of helper functions that can potentially be applied across multiple views.

## Acceptance Criteria
- [x] Graph card support or helper functions added. Determine whether support functions should be placed alongside graph component or inside helper file setup.
- [x] Helper functions should be testable to provide a baseline for future changes
- [x] It should have an approved design based loading state
- [x] It should have an approved design based error state

## Assumptions and Questions
- Investigate the Insights `skeleton` component for loading

- Date/time querystring format for beginning and ending... do we need to include the timestamp as part of the format?
- Date/time for beginning and ending... are those both inclusive of the returned result set?
- Sorting on the response. Will the list that's returned by the API already be sorted for us or do we need to do a fallback sort?
- Grouping within the data list that's returned by the API. Do we need to worry about entries that overlap with the same date?

### PR DESCRIPTION
**Pull request by priley86**
## What's included

* parse the Redux date and sockets inputs and generate chart data and
tooltips
* add error state and zeroed array
* loading state

Outstanding question - should a max-y value be set for non error state?

## Example

Loading desktop:
loading-desktop

Loading mobile:
loading-mobile

Loading complete:
loading-full

Error state (zeroed array), max-y set to 100, toast notification to come in future PR... : 
Screen Shot 2019-07-18 at 11 43 03 AM

## Updates issue/story

Closes #18

### CODE REVIEW

File: src/common/graphHelpers.js [hunk 1]
```diff
@@ -1,54 +1,97 @@
...
-    const clone = moment(startDate);
-    zeroedArray.push({ x: clone.add(i, 'days').format(chartDateFormat), y: 0 });
+  const endDateStartDateDiff = moment(endDate).diff(startDate, 'days');
+
+  // todo: convert "y" back towards a number if/when we handle "chartDomain.y = [0, 100]" within helpers
+  for (let i = 0; i <= endDateStartDateDiff; i++) {
+    const clonedStartDate = moment(startDate);
+    zeroedArray.push({
+      x: clonedStartDate.add(i, 'days').format(chartDateFormat),
+      y: '0'
```
Comment (priley86) [inline src/common/graphHelpers.js]: this is strange... happy to revisit after. Victory at it's  best...

File: src/common/graphHelpers.js [hunk 2]
```diff
@@ -1,54 +1,97 @@
...
-const convertGraphData = ({ usage, startDate, endDate, tSockectsOn, tFromPrevious }) => {
-  /**
-   * convert json usage report from this format:
-   * {cores: 56, date: "2019-06-01T00:00:00Z", instance_count: 28}
-   * to this format:
-   * { x: 'Jun 1', y: 56, label: '56 Sockets on Jun 1 \r\n +5 from previous day' },
-   */
-  if (usage === undefined || usage.length === 0) {
-    return zeroedUsageArray(startDate, endDate);
+  if (previousCores === null) {
```
Comment (priley86) [inline src/common/graphHelpers.js]: +1 for readability

File: src/components/rhelGraphCard/rhelGraphCard.js [hunk 3]
```diff
@@ -19,53 +19,58 @@ import { graphHelpers } from '../../common/graphHelpers';
...
+    const chartDomain = { x: [0, 31] };
+
     let chartData;
 
     if (error) {
-      // todo: show error toast?
-      chartData = graphHelpers.zeroedUsageArray(this.startDate, this.endDate);
+      // todo: evaluate show error toast
+      // todo: this only fires on error, not in the event an array with a valid zero length.
+      // todo: we may need to combine error and fulfilled checks and handle the x and y axis defaults in the helpers
```
Comment (priley86) [inline src/components/rhelGraphCard/rhelGraphCard.js]: sounds good...

File: src/components/rhelGraphCard/__tests__/__snapshots__/rhelGraphCard.test.js.snap [hunk 4]
```diff
@@ -246,125 +246,125 @@ exports[`RhelGraphCard Component should render a non-connected component: non-co
 exports[`RhelGraphCard Component should render multiple states: error shows zeroed bar values 1`] = `
 Object {
   "chartBarData": Array [
+    Object {
```
Comment (priley86) [inline src/components/rhelGraphCard/__tests__/__snapshots__/rhelGraphCard.test.js.snap]: i'm good after this bit is updated... i believe it should start w/ Jun 1 if Jun 1 UTC is passed.

Comment (cdcabrera) [reply src/components/rhelGraphCard/__tests__/__snapshots__/rhelGraphCard.test.js.snap]: ugh missed that

Comment (priley86) [reply src/components/rhelGraphCard/__tests__/__snapshots__/rhelGraphCard.test.js.snap]: ahh... yes, you are correct:

```
new Date('2019-06-01T00:00:00Z')
Fri May 31 2019 20:00:00 GMT-0400 (Eastern Daylight Time)
```

Ok, yes, if sourced from the server....shouldn't matter. I am guessing that Jest uses the system UTC time (so EST in my case), but not positive.

Maybe some way to mock this better in the future? FWIW, it's not obvious. Could also just set to something same day in UTC /U.S. time zones for being more obvious...

```
new Date('2019-06-01T12:00:00Z')
Sat Jun 01 2019 08:00:00 GMT-0400 (Eastern Daylight Time)
```

**Pull request merged**